### PR TITLE
INSTUI-2878 Upgrade storybook to 6.1.18

### PR DESCRIPTION
**Pull request by matyasf**
Uprade storybook to the latest version, hopefully this will help with the Chromatic issues. Also removed some exotic code.

### CODE REVIEW

File: packages/__examples__/.storybook/main.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+module.exports = {
+  stories: ['../stories.js']
+}
```
Comment (matyasf) [inline packages/__examples__/.storybook/main.js:27]: This is the new syntax, see https://storybook.js.org/blog/declarative-storybook-configuration/

File: packages/__examples__/.storybook/preview.js [hunk 2]
```diff
@@ -48,54 +35,9 @@ addParameters({
       name: 'instructure-ui',
       brandUrl: 'https://instructure.design',
       brandImage:
-        'https://instructure.design/icons-bea652a7d0546b99c7cba9d3b4a538ee/favicon-32x32.png'
+        'https://instructure.design/icons-bea652a7d0546b99c7cba9d3b4a538ee/favicon-32x32.png' // TODO this URL is broken
     }),
     showPanel: false,
     isFullscreen: false
   }
 })
```
Comment (matyasf) [inline packages/__examples__/.storybook/preview.js:43]: again, the new format, see https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#using-previewjs

File: packages/__examples__/stories.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+import { storiesOf } from '@storybook/react'
```
Comment (matyasf) [inline packages/__examples__/stories.js:25]: Although storiesOf API is legacy https://github.com/storybookjs/storybook/blob/next/lib/core/docs/storiesOf.md they do not seem to offer a new API to programatically load lots of stories, see https://github.com/storybookjs/storybook/issues/9828

File: packages/__examples__/.storybook/preview.js [hunk 4]
```diff
@@ -48,54 +35,9 @@ addParameters({
...
-
-configure(() => {
-  const examplesContext = require.context(
-    '../../',
-    true,
-    /^.*\/src\/.*\.examples\.js$/,
-    'lazy'
```
Comment (matyasf) [inline packages/__examples__/.storybook/preview.js:63]: This was one of the things causing issues with Storybook 6. We were loading our modules lazily with a promise (this is some barely documented Webpack feature) that caused Storybook to throw errors saying "Error: Cannot add a story when not configuring ...". I've set this to 'sync', see stories.js

**Comment by Brailor** [comment]
Somehow not all of our examples are shown with this new upgrade.
With the previous version of Storybook we had 50+ components, with this new version only 14 of them are being show when I run `yarn dev:examples`.
Could you check this out?

File: packages/ui-component-examples/lib/component-examples-loader.js [hunk 5]
```diff
@@ -96,21 +92,10 @@ config.maxExamples = Boolean(config.maxExamples) ? config.maxExamples : ${
         config.maxExamples
       }
 
-if (!config.renderPage) {
-  console.warn('A default is no longer provided for \`renderPage\`, you should supply it to the config instead')
-}
-
-if (!config.renderExample) {
-  console.warn('A default is no longer provided for \`renderExample\`, you should supply it to the config instead')
-}
```
Comment (matyasf) [inline packages/ui-component-examples/lib/component-examples-loader.js:105]: here they likely wanted to check the opposite (`if (config.renderExample)`), see https://github.com/instructure/instructure-ui/commit/3abd6cac709de54e1be2eee3b5a6c22c0bdadb7b#diff-f82b6242f20fe2b1c9d6cf5c3522ad1c9463b3a9d8bd703c307050df0f079007

**Review by Brailor — APPROVED** [review]
Great work!

**Review by ke1k0 — COMMENTED** [review]
Img is still throwing error: ```Warning: Failed prop type: Invalid prop `src` of type `object` supplied to `Img`, expected `string`.
    in Img (created by unboundStoryFn)
    in unboundStoryFn
    in ErrorBoundary```
    
Also, there is a conflict in yarn.lock

**Comment by matyasf** [comment]
> Img is still throwing error: `` Warning: Failed prop type: Invalid prop `src` of type `object` supplied to `Img`, expected `string`. in Img (created by unboundStoryFn) in unboundStoryFn in ErrorBoundary ``
> 
> Also, there is a conflict in yarn.lock

hmm I fixed this in one place, then there must be other places where the `src` of the `img` is not set correctly

**Comment by matyasf** [comment]
> Img is still throwing error: `` Warning: Failed prop type: Invalid prop `src` of type `object` supplied to `Img`, expected `string`. in Img (created by unboundStoryFn) in unboundStoryFn in ErrorBoundary ``
> 
> Also, there is a conflict in yarn.lock

I solved the conflict and fixed all the errors that components were throwing

**Pull request merged**